official/resnet/keras/resnet_model.py

```diff
@@ -0,0 +1,468 @@
+  x = tf.keras.layers.BatchNormalization(axis=bn_axis, name='bn_conv1',
+                                         momentum=BATCH_NORM_DECAY,
+                                         epsilon=BATCH_NORM_EPSILON)(
+                                             x, training=training)
+  x = tf.keras.layers.Activation('relu')(x)
+  x = tf.keras.layers.MaxPooling2D((3, 3), strides=(2, 2))(x)
+
+  x = conv_block(x, 3, [64, 64, 256], stage=2, block='a', strides=(1, 1),
+                 training=training)
+  x = identity_block(x, 3, [64, 64, 256], stage=2, block='b',
+                     training=training)
+  x = identity_block(x, 3, [64, 64, 256], stage=2, block='c',
+                     training=training)
+
+  x = conv_block(x, 3, [128, 128, 512], stage=3, block='a',
+                 training=training)
+  x = identity_block(x, 3, [128, 128, 512], stage=3, block='b',
+                     training=training)
+  x = identity_block(x, 3, [128, 128, 512], stage=3, block='c',
+                     training=training)
+  x = identity_block(x, 3, [128, 128, 512], stage=3, block='d',
+                     training=training)
+
+  x = conv_block(x, 3, [256, 256, 1024], stage=4, block='a',
+                 training=training)
+  x = identity_block(x, 3, [256, 256, 1024], stage=4, block='b',
+                     training=training)
+  x = identity_block(x, 3, [256, 256, 1024], stage=4, block='c',
+                     training=training)
+  x = identity_block(x, 3, [256, 256, 1024], stage=4, block='d',
+                     training=training)
+  x = identity_block(x, 3, [256, 256, 1024], stage=4, block='e',
+                     training=training)
+  x = identity_block(x, 3, [256, 256, 1024], stage=4, block='f',
+                     training=training)
+
+  x = conv_block(x, 3, [512, 512, 2048], stage=5, block='a',
+                 training=training)
+  x = identity_block(x, 3, [512, 512, 2048], stage=5, block='b',
+                     training=training)
+  x = identity_block(x, 3, [512, 512, 2048], stage=5, block='c',
+                     training=training)
+
+  if include_top:
+    x = tf.keras.layers.AveragePooling2D((7, 7), name='avg_pool')(x)
+    x = tf.keras.layers.Flatten()(x)
+    x = tf.keras.layers.Dense(classes, activation='linear', name='fc1000')(x)
+  else:
+    if pooling == 'avg':
+      x = tf.keras.layers.GlobalAveragePooling2D()(x)
+    elif pooling == 'max':
+      x = tf.keras.layers.GlobalMaxPooling2D()(x)
+    else:
+      warnings.warn('The output shape of `ResNet50(include_top=False)` '
+                    'has been changed since Keras 2.2.0.')
+
+  # Ensure that the model takes into account
+  # any potential predecessors of `input_tensor`.
+  if input_tensor is not None:
+    inputs = tf.keras.engine.get_source_inputs(input_tensor)
+  else:
+    inputs = img_input
+  # Create model.
+  model = tf.keras.models.Model(inputs, x, name='resnet50')
+
+  # Load weights.
+  if weights == 'imagenet':
```

Comment:
Is this block needed? In particular, are the weights in WEIGHTS_PATH and WEIGHTS_PATH_NO_TOP in sync with the precise model definition in this file (I thought they were based on the model definition of `keras.applications.ResNet50` which is slightly different)?

Since this file implements a distinct model, I'd be tempted to not worry about loading weights from a different model and reduce the number of untested codepaths.